File: swift-backup/Design/Theme.swift
import SwiftUI

struct Theme {
    // Spacing
    static let spacingXS: CGFloat = 4
    static let spacingS: CGFloat = 8
    static let spacingM: CGFloat = 12
    static let spacingL: CGFloat = 16
    static let spacingXL: CGFloat = 20
    static let spacingXXL: CGFloat = 24
    
    // Corner Radius
    static let cornerRadiusS: CGFloat = 8
    static let cornerRadiusM: CGFloat = 12
    static let cornerRadiusL: CGFloat = 16
    static let cornerRadiusXL: CGFloat = 24
    
    // Shadows
    static let shadowSmall = Shadow(radius: 4, opacity: 0.08)
    static let shadowMedium = Shadow(radius: 8, opacity: 0.12)
    static let shadowLarge = Shadow(radius: 12, opacity: 0.16)
    
    // Colors
    static let primaryColor = Color.purple
    static let secondaryColor = Color.teal
    static let accentColor = Color.orange
    
    // Event Colors
    static let eventColors: [Color] = [
        Color.orange.opacity(0.2),
        Color.purple.opacity(0.2),
        Color.blue.opacity(0.2),
        Color.pink.opacity(0.2),
        Color.yellow.opacity(0.2),
        Color.green.opacity(0.2),
        Color.indigo.opacity(0.2)
    ]
    
    static let eventColorBorders: [Color] = [
        Color.orange,
        Color.purple,
        Color.blue,
        Color.pink,
        Color.yellow,
        Color.green,
        Color.indigo
    ]
}

struct Shadow {
    let radius: CGFloat
    let opacity: Double
}

extension View {
    func themeShadow(_ shadow: Shadow) -> some View {
        self.shadow(color: .black.opacity(shadow.opacity), radius: shadow.radius, x: 0, y: 2)
    }
}

File: src/theme/Theme.ts
// Theme constants for React Native styling
export const Theme = {
  // Spacing
  spacingXS: 4,
  spacingS: 8,
  spacingM: 12,
  spacingL: 16,
  spacingXL: 20,
  spacingXXL: 24,

  // Corner Radius
  cornerRadiusS: 8,
  cornerRadiusM: 12,
  cornerRadiusL: 16,
  cornerRadiusXL: 24,

  // Shadows
  shadowSmall: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.08,
    shadowRadius: 4,
    elevation: 4,
  },
  shadowMedium: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.12,
    shadowRadius: 8,
    elevation: 8,
  },
  shadowLarge: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.16,
    shadowRadius: 12,
    elevation: 12,
  },

  // Colors
  primaryColor: '#8B5CF6', // purple
  secondaryColor: '#14B8A6', // teal
  accentColor: '#FB923C', // orange

  // Event Colors (with transparency)
  eventColors: [
    '#FB923C33', // orange with opacity
    '#8B5CF633', // purple with opacity
    '#3B82F633', // blue with opacity
    '#EC489633', // pink with opacity
    '#FDE04733', // yellow with opacity
    '#10B98133', // green with opacity
    '#6366F133', // indigo with opacity
  ],

  // Event Border Colors
  eventColorBorders: [
    '#FB923C', // orange
    '#8B5CF6', // purple
    '#3B82F6', // blue
    '#EC4899', // pink
    '#FDE047', // yellow
    '#10B981', // green
    '#6366F1', // indigo
  ],

  // Text Colors
  textPrimary: '#111827',
  textSecondary: '#6B7280',
  textTertiary: '#9CA3AF',

  // Background Colors
  backgroundPrimary: '#FFFFFF',
  backgroundSecondary: '#F9FAFB',
  backgroundTertiary: '#F3F4F6',
};

export type ShadowStyle = typeof Theme.shadowSmall;
